sdk/log/record.go

```diff
@@ -142,8 +181,103 @@ func (r *Record) AddAttributes(attrs ...log.KeyValue) {
 	r.back = append(r.back, attrs[i:]...)
 }
 
+func (r *Record) compactAttr() {
```

Comment:
it would be nice to have a comment for this function.

Reply:
Updated.